MapboxMobileEvents/Categories/NSUserDefaults+MMEConfiguration.m

```diff
@@ -290,20 +290,30 @@ - (NSString *)mme_userAgentString {
             NSBundle.mme_mainBundle.mme_bundleVersionString,
             NSBundle.mme_mainBundle.bundleIdentifier,
             NSBundle.mme_mainBundle.infoDictionary[(id)kCFBundleVersionKey]];
+            NSBundle.mme_mainBundle.infoDictionary[(id)kCFBundleNameKey] ?: NSBundle.mme_mainBundle.bundlePath.lastPathComponent.stringByDeletingPathExtension,
         
-        // check all loaded frameworks for mapbox frameworks
-        for (NSBundle *loaded in NSBundle.allFrameworks) {
-            if (loaded != NSBundle.mme_mainBundle
-            && loaded.bundleIdentifier
-            && [loaded.bundleIdentifier rangeOfString:@"mapbox" options:NSCaseInsensitiveSearch].location != NSNotFound) {
-                NSString *uaFragment = [NSString stringWithFormat:@" %@/%@ (%@; v%@)",
-                    loaded.infoDictionary[(id)kCFBundleNameKey],
-                    loaded.mme_bundleVersionString,
-                    loaded.bundleIdentifier,
-                    loaded.infoDictionary[(id)kCFBundleVersionKey]];
-                userAgent = [userAgent stringByAppendingString:uaFragment];
+        // check all loaded frameworks for mapbox frameworks, record thier bundleIdentifier
+        // check all loaded frameworks for mapbox frameworks, record their bundleIdentifier
+        NSMutableSet *loadedMapboxBundleIds = NSMutableSet.new;
+        for (NSBundle *loaded in [NSBundle.allFrameworks arrayByAddingObjectsFromArray:NSBundle.allBundles]) {
+            if (loaded.bundleIdentifier
+             && loaded.bundleIdentifier != NSBundle.mme_mainBundle.bundleIdentifier
+             && [loaded.bundleIdentifier rangeOfString:@"mapbox" options:NSCaseInsensitiveSearch].location != NSNotFound) {
+                [loadedMapboxBundleIds addObject:loaded.bundleIdentifier];
             }
         }
+        
+        // sort the bundleIdentifiers, then use them to build the User-Agent string
+        NSArray *sortedBundleIds = [loadedMapboxBundleIds sortedArrayUsingDescriptors:@[[NSSortDescriptor sortDescriptorWithKey:nil ascending:YES]]];
+        for (NSString *bundleId in sortedBundleIds) {
+            NSBundle *loaded = [NSBundle bundleWithIdentifier:bundleId];
+            NSString *uaFragment = [NSString stringWithFormat:@" %@/%@ (%@; v%@)",
+                loaded.infoDictionary[(id)kCFBundleNameKey] ?: loaded.bundlePath.lastPathComponent.stringByDeletingPathExtension,
```

Comment:
Why delete the extension?

Reply:
AFAIK this is to prevent a `null` value being passed through. @alfwatt for more context

Reply:
/long/path/to/MapboxMobileEvents.framework → MapboxMobileEvents

Reply:
Which is typically what kCFBundleNameKey would be in English, the key exists so it can be localized